internal/nvsandboxutils/gpu-resources.go

```diff
@@ -0,0 +1,74 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package nvsandboxutils
+
+import (
+	"strings"
+)
+
+import "C"
+
+type GpuResource struct {
+	Version uint32
+}
+
+//	type GpuFileInfo_v1 struct {
+//		Next        *GpuFileInfo_v1
+//		FileType    uint32
+//		FileSubType uint32
+//		Module      uint32
+//		Flags       uint32
+//		FilePath    *int8
+//	}
+type GpuFileInfo struct {
+	Path    string
+	Type    FileType
+	SubType FileSystemSubType
+	Module  FileModule
+	Flags   FileFlag
+}
+
+func (l *library) GetGpuResource(uuid string) ([]GpuFileInfo, Ret) {
```

Comment:
One thing that we have considered doing in `go-nvml` as well (see https://github.com/NVIDIA/go-nvml/pull/125) is returning an `error` instead of a `Ret` value. This cleans up most standard interactions with the library in the calling code.

I'm happy to take this on as a follow-up though if we don't want to do it as part of the initial commits.

Reply:
I see, it does look cleaner. I can create a follow-up patch for those changes once the initial commits for the library are in.